compiler/optimizer/LoopReplicator.cpp

```diff
@@ -729,7 +748,7 @@ bool TR_LoopReplicator::heuristics(LoopInfo *lInfo)
    if (!lInfo->_replicated)
       dumpOptDetails(comp(), "no side entrance found into trace; no replication will be performed\n");
 
-   return true;
+   return lInfo->_replicated;
```

Comment:
These are functional changes (this and the other return further down - could you just document here if you observed a problem form this and what this change did since it could have an impact on perf etc and I'd like to understand the problem case.

Reply:
The return value of `heuristics` only affects trace messages and the result of `TR_LoopReplicator::perform`. The result of `perform` seems to be unused. (It feeds into the result of `Optimizer::performOptimization`, which is itself unused except to feed into itself recursively.)

Reply:
Ok sounds good.

Reply:
Echoing Devin, this does not affect performance, only enhancing correctness by making the caller of heuristics methods generating the correct traceMsg. I can add some comment in the code.

Reply:
code comment not required - just wanted to document the justification of the change in the PR - I'm fine with it especially in light of the comments from @jdmpapin